frontend/src/common/queries/filter.ts

```diff
@@ -684,48 +720,61 @@ function processCollectionResponse(
+  dataset: DatasetResponse
+): ProcessedDatasetResponse {
+  // Build up value to facilitate ontology-aware tissue filtering.
+  const tissueCalculated = [
+    ...tagAncestorsAsInferred(dataset.tissue_ancestors),
+    ...tagOntologyTermsAsExplicit(dataset.tissue),
+  ];
+  return {
+    ...dataset,
+    tissueCalculated,
+  };
+}
+
 /**
  * Add defaults for missing filterable values, e.g. convert missing ontology values to empty array.
  * Remove any ethnicity values on the deny list.
- * @param dataset - Dataset to check for missing values.
+ * @param datasetResponse - Dataset to check for missing values.
  * @returns Corrected dataset response.
  */
-function sanitizeDataset(dataset: DatasetResponse): DatasetResponse {
-  return Object.values(CATEGORY_KEY).reduce(
-    (accum: DatasetResponse, categoryKey: CATEGORY_KEY) => {
-      // Check for fields that don't require sanitizing.
-      if (
-        categoryKey === CATEGORY_KEY.CELL_COUNT ||
-        categoryKey === CATEGORY_KEY.MEAN_GENES_PER_CELL ||
-        categoryKey === CATEGORY_KEY.PUBLICATION_AUTHORS ||
-        categoryKey === CATEGORY_KEY.PUBLICATION_DATE_VALUES
-      ) {
-        return accum;
-      }
-
-      if (categoryKey === CATEGORY_KEY.DEVELOPMENT_STAGE_ANCESTORS) {
-        accum.development_stage_ancestors =
-          dataset.development_stage_ancestors ?? [];
-        return accum;
-      }
-
-      if (categoryKey === CATEGORY_KEY.TISSUE_ANCESTORS) {
-        accum.tissue_ancestors = dataset.tissue_ancestors ?? [];
-        return accum;
-      }
-
-      if (categoryKey === CATEGORY_KEY.ETHNICITY) {
-        accum.ethnicity = (dataset.ethnicity ?? []).filter(
-          (ethnicity) => !ETHNICITY_DENY_LIST.includes(ethnicity.label)
-        );
-        return accum;
-      }
-
-      accum[categoryKey] = dataset[categoryKey] ?? [];
-      return accum;
-    },
-    { ...dataset }
+function sanitizeDatasetResponse(
+  datasetResponse: DatasetResponse
+): DatasetResponse {
+  const sanitizedDatasetResponse = { ...datasetResponse };
+
+  sanitizedDatasetResponse.ethnicity = (datasetResponse.ethnicity ?? []).filter(
+    (ethnicity) => !ETHNICITY_DENY_LIST.includes(ethnicity.label)
   );
+
+  sanitizedDatasetResponse.assay = datasetResponse.assay ?? [];
+
+  sanitizedDatasetResponse.cell_type = datasetResponse.cell_type ?? [];
+
+  sanitizedDatasetResponse.cell_type_ancestors =
+    datasetResponse.cell_type_ancestors ?? [];
+
+  sanitizedDatasetResponse.development_stage_ancestors =
+    datasetResponse.development_stage_ancestors ?? [];
+
+  sanitizedDatasetResponse.disease = datasetResponse.disease ?? [];
+
+  sanitizedDatasetResponse.organism = datasetResponse.organism ?? [];
+
+  sanitizedDatasetResponse.sex = datasetResponse.sex ?? [];
+
+  sanitizedDatasetResponse.tissue = datasetResponse.tissue ?? [];
+
+  sanitizedDatasetResponse.tissue_ancestors =
+    datasetResponse.tissue_ancestors ?? [];
+
+  return sanitizedDatasetResponse;
```

Comment:
Timmy! I actually really struggled with the typing in this function, to the point where I split it out into the current format of individual checks. Let me create a ticket to revisit this.

Reply:
Added ticket #3260.

Reply:
Yeah I even tried a similar example to see if it works and it does 🤯 

```ts
interface A {
  a: string;
  b: string;
}

const keys: (keyof A)[] = ["a", "b"];

const a: A = { a: "hi", b: "ho" };
const b: A = { ...a };

for (const key of keys) {
  b[key] = a[key]; // no error
}
```

Probably need to dig a little more to compare/contrast there 😆 So interesting!

I did notice that if `interface A` has an optional key then the assignment will throw an error, but it's not the same `never` error 🤔 

<img width="639" alt="Screen Shot 2022-09-13 at 10 07 39 AM" src="https://user-images.githubusercontent.com/6309723/189964114-c9776a8e-078c-4f7b-86a1-597f5ae586c8.png">

Definitely a low priority thing, because it could be a TypeScript bug too 🤷 AHAHAHA